### PR TITLE
Upgrade ruby version to 2.6.x

### PR DESCRIPTION
**Pull request by abenbachir**
Benefits of upgrading is more about CPU improvements, most importantly is the JIT feature that was introduced in 2.6.
Deprecation of SSL 1.0.0, required system installation is 1.0.1 onward and 1.1.*
Build requirements: GCC 4.7+, OpenSSL 1.0.1+

### CODE REVIEW

File: source/code/plugins/mysql_workload_lib.rb [hunk 1]
```diff
@@ -185,7 +185,7 @@ def get_server_stats
 
     # add MySQL Worst Case RAM Usage
     server_max_ram= Hash.new
-    value = 0 #reset
+    value = 0.0 #reset
```
Comment (simathih) [inline source/code/plugins/mysql_workload_lib.rb]: is 0.0 a ruby 2.6  thing?

**Comment by simathih** [comment]
how will we test this? will shell bundle testing cover everything?

File: build/Makefile [hunk 2]
```diff
@@ -124,10 +124,12 @@ SYSTEST_CONFIG := $(BASE_DIR)/test/config/systest.conf
 .PHONY: all clean clean-ruby distclean clean-status kit
 .PHONY: tests test omstest unittest systemtest systemtestrb
 
-all : $(OMI_LIBRARY_LIB_BASE) $(DSC_TARGET_DIR) $(RUBY_DEST_DIR) $(IN_PLUGINS_LIB) sepolicy kit
+all : $(OMI_LIBRARY_LIB_BASE) $(DSC_TARGET_DIR) $(RUBY_DEST_DIR) $(IN_PLUGINS_LIB) kit
         # After a successful make, undo the Ruby patches for sequential builds from old branches
 	@echo "Cleaning patched Ruby files..."
-	cd $(BASE_DIR)/source/ext/ruby; git checkout -- ext/openssl/ossl_ssl.c ext/openssl/extconf.rb
```
[the remark this answers is not in the document]
Comment (abenbachir) [reply build/Makefile]: an old patch of openssl could not be applied, check ./build/configure

File: source/code/plugins/mysql_workload_lib.rb (hunk 1 above)
Comment (abenbachir) [reply source/code/plugins/mysql_workload_lib.rb]: This is json improvements in Ruby 2.6
an integer will be dumped as "0" not "0.0".

**Comment by abenbachir** [comment]
> how will we test this? will shell bundle testing cover everything?

Yes, any issue should be able to catch by shell bundle testing.
I already did some testing on Redhat after fixing the SSL-fips issue.

File: build/Makefile [hunk 3]
```diff
@@ -124,10 +124,12 @@ SYSTEST_CONFIG := $(BASE_DIR)/test/config/systest.conf
 .PHONY: all clean clean-ruby distclean clean-status kit
 .PHONY: tests test omstest unittest systemtest systemtestrb
 
-all : $(OMI_LIBRARY_LIB_BASE) $(DSC_TARGET_DIR) $(RUBY_DEST_DIR) $(IN_PLUGINS_LIB) sepolicy kit
```
[the remark this answers is not in the document]
Comment (abenbachir) [reply build/Makefile]: fixed

**Pull request merged**